legend-engine-external-shared-format-model/src/main/java/org/finos/legend/engine/external/shared/format/model/ExternalFormatExtension.java

```diff
@@ -64,78 +62,37 @@
-
     /**
      * Called to convert a compiled schema detail back to its textual (schema-specific grammar) form
      */
     String metamodelToText(Metamodel schemaDetail, PureModel pureModel);
 
     /**
-     * Determines whether this extension supports model generation.
+     * Called as part of Binding compilation to verify that the model and schema match such that serialization and deserialization
+     * can occur.  This provides early warning to the user as they change the schema and model of changes that would break data flows.
+     * The compilation fails if the result returned is a meta_external_shared_format_binding_validation_FailedCorrelation.
      */
-    default boolean supportsModelGeneration()
+    default Root_meta_external_shared_format_binding_validation_BindingDetail bindDetails(Root_meta_external_shared_format_binding_Binding binding, CompileContext context)
     {
-        return false;
-    }
+        Root_meta_external_shared_format_ExternalFormatContract<Metamodel> externalFormatContract = getExternalFormatContract(context.pureModel);
 
-    /**
-     * Provides the properties used to configure model generation.
-     */
-    default RichIterable<? extends Root_meta_pure_generation_metamodel_GenerationParameter> getModelGenerationProperties(PureModel pureModel)
-    {
-        if (this.supportsModelGeneration())
+        if (externalFormatContract._externalFormatBindingValidator() == null)
         {
-            throw new IllegalStateException("Must supply properties if supporting generation");
+            throw new EngineException("Content Type does not support binding validation", SourceInformation.getUnknownSourceInformation(), EngineErrorType.COMPILATION);
         }
-        return Lists.mutable.empty();
-    }
 
-    /**
-     * Called when an external schema of this format is used to generate a PURE model.
-     */
-    Root_meta_external_shared_format_binding_Binding generateModel(Root_meta_external_shared_format_metamodel_SchemaSet schema, ModelGenConfig config, PureModel pureModel);
-
-    /**
-     * Determines whether this extension supports schema generation.
-     */
-    default boolean supportsSchemaGeneration()
-    {
-        return false;
-    }
-
-    /**
-     * Provides the properties used to configure model generation.
-     */
-    default RichIterable<? extends Root_meta_pure_generation_metamodel_GenerationParameter> getSchemaGenerationProperties(PureModel pureModel)
-    {
-        if (this.supportsSchemaGeneration())
-        {
-            throw new IllegalStateException("Must supply properties if supporting generation");
-        }
-        return Lists.mutable.empty();
+        return (Root_meta_external_shared_format_binding_validation_BindingDetail) CoreGen.evaluate(context.getExecutionSupport(), externalFormatContract._externalFormatBindingValidator(), binding);
```

Comment:
I think we rarely use CoreGen.evaluate. May be introduce a function in PURE which is called with the contract and binding like we do in other places (plan to platform binding etc).

Reply:
Any reason we want to avoid CoreGen.evaluate, I am using at other places too and it might add clutter to have such functions